src/mg/PAMI/Message/IncomingMessage.php

```diff
@@ -81,6 +87,33 @@ public function getRawContent()
     }
 
     /**
+     * Returns the channel variables for all reported channels.
+     * https://github.com/marcelog/PAMI/issues/85
+     *
+     * The channel names will be lowercased.
+     *
+     * @return array
+     */
+    public function getAllChannelVariables()
+    {
+      return $this->channelVariables;
+    }
+
+    /**
+     * Returns the channel variables for the given channel.
+     * https://github.com/marcelog/PAMI/issues/85
+     *
+     * @param string $channel Channel name. If not given, will return variables
+     * for the "current" channel.
+     *
+     * @return array
+     */
+    public function getChannelVariables($channel = 'default')
```

Comment:
This contains a bug for channels with the second syntax. As you can see in the test case, there are a few channel variables set:

```
            # Included 'Channel' for clarity
            'Channel: Local/0@pbx_dial_callroute_to_endpoint-00000008;2',
            'ChanVariable: var1',
            'ChanVariable: var2=v2',
            'ChanVariable(Local/0@pbx_dial_callroute_to_endpoint-00000008;2): var1=value1',
            'ChanVariable(Local/0@pbx_dial_callroute_to_endpoint-00000008;2): var2=value2',
            'ChanVariable(Local/0@pbx_dial_callroute_to_endpoint-00000008;2): var3=value3',
            'ChanVariable(SIP/jw1034-00000010): var12=value12',
            'ChanVariable(SIP/jw1034-00000010): var22=value22',
            'ChanVariable(SIP/jw1034-00000010): var32=value32',
```

Since the channel name is `Local/0@pbx_dial_callroute_to_endpoint-00000008;2` here, not only the `var1` and `var2` variables should be returned, but also the `var1`, `var2` and `var3` variables of the channel with name `Local/0@pbx_dial_callroute_to_endpoint-00000008;2`.

In fact, I think there should be two test cases: one for the first syntax (`ChanVariable: var2=v2`) and one for the second syntax (`ChanVariable(SIP/jw1034-00000010): var12=value12`).

In both cases, when using just `::getVariables()`, you should get all the variables from the current channel.